backend/dal/Services/Concrete/ParcelService.cs

```diff
@@ -212,6 +212,20 @@ public Parcel Add(Parcel parcel)
         /// <exception type="KeyNotFoundException">Entity does not exist in the datasource.</exception>
         /// <returns></returns>
         public Parcel Update(Parcel parcel)
+        {
+            parcel = PendingUpdate(parcel);
+            this.Context.SaveChanges();
+            this.Context.CommitTransaction();
+            return parcel;
+        }
+
+        /// <summary>
+        /// Update the specified parcel in the datasource, but do not commit the transaction.
+        /// </summary>
+        /// <param name="parcel"></param>
+        /// <exception type="KeyNotFoundException">Entity does not exist in the datasource.</exception>
+        /// <returns></returns>
+        public Parcel PendingUpdate(Parcel parcel)
```

Comment:
The naming here could probably change, but I want to be able to run this update logic without committing so that I can do the building -> associated parcel updates in a single transaction so that I can rollback everything if there is an error.